Doc/c-api/list.rst

```diff
@@ -51,6 +51,23 @@ List Objects
    ``len(list)`` on a list object.
 
 
+.. c:function:: int PyList_Clear(PyObject *list)
+
+   Remove all items from *list*. Similar to:
+   ``PyList_SetSlice(list, 0, PY_SSIZE_T_MAX, NULL)``.
```

Comment:
Should we way anything about whether the underlying storage is resized?

Reply:
https://docs.python.org/dev/c-api/list.html documentation doesn't say anything about memory allocation, overallocation or freeing the internal storage. I would prefer to not start to document internals, unless there is a good reasons to document the exact behavior of PyList_Clear() related to memory handling.